### PR TITLE
mstart: make some mstart env variables configurable 

### PR DESCRIPTION
**Pull request by nizamial09**
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/mrun [hunk 1]
```diff
@@ -6,7 +6,7 @@ root=`dirname $0`
 run_name=$1
 command=$2
 CEPH_BIN=""
-CEPH_CONF_PATH="/etc/ceph/"
+[ -z "$CEPH_CONF_PATH" ] && CEPH_CONF_PATH="/etc/ceph/"
```
[the remark this answers is not in the document]
Comment (nizamial09) [reply src/mrun]: done, thanks

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
@epuertat I've made the changes. I've also isolated the client logs in a separate commit as per your [comment here](https://github.com/ceph/ceph-build/pull/2030#issuecomment-1201488409)

File: src/vstart.sh [hunk 2]
```diff
@@ -36,13 +36,14 @@ if [ -n "$VSTART_DEST" ]; then
...
-    CEPH_BIN=${PWD}/bin
-    CEPH_LIB=${PWD}/lib
+    CEPH_BIN=${CEPH_BIN:-${PWD}/bin}
+    CEPH_LIB=${CEPH_LIB:-${PWD}/lib}
 
     CEPH_CONF_PATH=$VSTART_DEST
     CEPH_DEV_DIR=$VSTART_DEST/dev
     CEPH_OUT_DIR=$VSTART_DEST/out
     CEPH_ASOK_DIR=$VSTART_DEST/asok
+    CEPH_OUT_CLIENT_DIR=$CEPH_OUT_DIR/client
```
Comment (epuertat) [inline src/vstart.sh:46]: As this change will also impact non-dashboard users, would about keeping the default setting (`$CEPH_OUT_DIR`), but, allowing users to tweak it (`$CEPH_OUT_CLIENT_DIR`)?
```suggestion
    CEPH_OUT_CLIENT_DIR=${CEPH_OUT_CLIENT_DIR:-$CEPH_OUT_DIR}
```

**Review by epuertat — APPROVED** [review]

File: src/vstart.sh (hunk 2 above)
Comment (nizamial09) [reply src/vstart.sh:46]: then that would mean setting this directory on our jobs explicitly right?

Comment (epuertat) [reply src/vstart.sh:46]: yeah! the idea was not to change the legacy default behaviour of vstart, but allow to customize it when required.

Comment (nizamial09) [reply src/vstart.sh:46]: fair enough. I'll update it.

File: src/vstart.sh [hunk 3]
```diff
@@ -131,6 +132,7 @@ fi
 [ -z "$CEPH_ASOK_DIR" ] && CEPH_ASOK_DIR="$CEPH_DIR/asok"
 [ -z "$CEPH_RGW_PORT" ] && CEPH_RGW_PORT=8000
 [ -z "$CEPH_CONF_PATH" ] && CEPH_CONF_PATH=$CEPH_DIR
+CEPH_OUT_CLIENT_DIR=${CEPH_OUT_CLIENT_DIR:-$CEPH_OUT_DIR}
```
Comment (nizamial09) [inline src/vstart.sh:135]: this guy can be bullied by others just for being different

File: src/vstart.sh (hunk 2 above)
Comment (nizamial09) [reply src/vstart.sh]: done! can you also check the `export CEPH_OUT_CLIENT_DIR` that I am doing is correct. and do I need to modify it in any other tests?

**Comment by nizamial09** [comment]
api logs
Screenshot from 2022-08-03 23-00-46

**Comment by nizamial09** [comment]
and dashboard e2es
Screenshot from 2022-08-04 12-08-03

**Comment by nizamial09** [comment]
jenkins test make check

File: src/vstart.sh (hunk 3 above)
Comment (pereman2) [reply src/vstart.sh:135]: survival of the fittest

**Review by pereman2 — APPROVED** [review]

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test windows

**Comment by nizamial09** [comment]
jenkins test make check arm64

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by nizamial09** [comment]
jenkins test api

**Comment by nizamial09** [comment]
jenkins test make check

**Comment by nizamial09** [comment]
jenkins test windows

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Pull request merged**